packages/polaris-viz/src/components/Labels/utilities/endWordTruncate.ts

```diff
@@ -30,5 +30,9 @@ export function endWordTruncate({word, targetWidth, characterWidths}: Props) {
     newWord = `${newWord}${ELLIPSIS}`;
   }
 
+  if (newWord.length === word.length - 1) {
```

Comment:
I don't know if it's worth introducing this complexity (yet, or ever), but how do we want this to handle unicode characters where a single visible character may not correspond to a length of `1`:

```javascript
function getGraphemeCount(str) {
  const segmenter = new Intl.Segmenter("en-US", { granularity: "grapheme" });
  return [...segmenter.segment(str)].length;
}

function getCodePointsCount(str) {
  return [...str].length;
}

console.log(getGraphemeCount("안녕하세요")); // 5
console.log(getCodePointsCount("안녕하세요")); // 5
console.log("안녕하세요".length); // 5

console.log(getGraphemeCount("👨‍👩‍👧‍👧")); // 1
console.log(getCodePointsCount("👨‍👩‍👧‍👧")); // 7
console.log("👨‍👩‍👧‍👧".length); // 11
```

Reply:
Interesting! I have seen emojis in product titles so this is probably actually a problem. I will open an issue